### PR TITLE
tests: tests for new wallet service facade methods

### PR DESCRIPTION
**Pull request by andreabadesso**
## Motivation

This PR solves https://github.com/HathorNetwork/hathor-wallet-lib/issues/351

It also contains a fix for the `changePin` method

### Acceptance Criteria
- We should have a test for the `executeGenerateWallet` method
- We should have a test for the `generateCreateWalletAuthData` method
- We should have a test for the  `getAcctPathXprivKey` method
- We should have a test to make sure that changing the pin will update the `acctPathXprivkey` correctly on storage

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/wallet.js [hunk 1]
```diff
@@ -918,7 +918,7 @@ const wallet = {
 
       newAccessData = {
         ...newAccessData,
-        acctPathMainKey: encryptedAccPathMainKey,
+        acctPathMainKey: encryptedAccPathMainKey.encrypted.toString(),
```
Comment (r4mmer) [inline src/wallet.js:921]: Is this a bug we found on this PR?

File: src/wallet/wallet.ts [hunk 2]
```diff
@@ -322,8 +322,8 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     let privKeyAccountPath: bitcore.HDPrivateKey;
     let authDerivedPrivKey: bitcore.HDPrivateKey;
 
-    const now = new Date();
-    const timestampNow = Math.floor(now.getTime() / 1000); // in seconds
+    const now = Date.now();
+    const timestampNow = Math.floor(now / 1000); // in seconds
```
Comment (r4mmer) [inline src/wallet/wallet.ts:326]: Was this changed so we could mock `Date.now()` on the tests?

**Review by r4mmer — APPROVED** [review]

File: src/wallet/wallet.ts (hunk 2 above)
Comment (andreabadesso) [reply src/wallet/wallet.ts:326]: Exactly

File: src/wallet.js (hunk 1 above)
Comment (andreabadesso) [reply src/wallet.js:921]: Yes it is, that's the bug I described on the PR description:

`It also contains a fix for the changePin method`

File: __tests__/wallet/wallet.test.ts [hunk 3]
```diff
@@ -227,3 +236,50 @@ test('getTxBalance', async () => {
...
+  const firstAddress = 'WR1i8USJWQuaU423fwuFQbezfevmT4vFWX';
+  const xpubAddress = 'WdSD7aytFEZ5Hp8quhqu3wUCsyyGqcneMu';
+  const authXpubAddress = 'WbjNdAGBWAkCS2QVpqmacKXNy8WVXatXNM';
+
+  const privKeyAccountPath: HDPrivateKey = new HDPrivateKey(xprivAccountPath);
+  const authDerivedPrivKey: HDPrivateKey = new HDPrivateKey(authXprivKey);
+
+  const xpubkeySignature = wallet.signMessage(privKeyAccountPath, timestampNow, walletId);
+  const authXpubkeySignature = wallet.signMessage(authDerivedPrivKey, timestampNow, walletId);
+  const xpubChangeDerivation = walletUtils.xpubDeriveChild(xpub, 0);
```
Comment (pedroferreira1) [inline __tests__/wallet/wallet.test.ts:274]: Why do you need this?

**Review by pedroferreira1 — APPROVED** [review]

File: __tests__/wallet/wallet.test.ts (hunk 3 above)
Comment (andreabadesso) [reply __tests__/wallet/wallet.test.ts]: Those 3 lines are leftovers, thanks

Removed in 7f2661426d9e44902539e6fc8733f39b8747bd99

**Pull request merged**